### ISSUE TITLE
Generate database for 0.17.xx

### ISSUE THREAD

**Issue by mvy** [kevinta893/factorio-recipes-json#5]
Hello, 

There is no mention of how you generated the database. Maybe it's common knowledge, but if you could give a pointer or two in the README it would be much appreciated. 

Unless you want to be super kind and provide it directly, but with all the frequent updates it's probably not advised. 

Thanks

**Comment by kevinta893** [comment]
Ah, I should have put it down in my readme since I haven't been updating this database with the new game updates recently. But how I've generated this database was actually by webscraping the [Factorio Wiki](https://wiki.factorio.com/). I used the chrome plugin [Webscraper](https://www.webscraper.io/) for this. Unfortunately I don't have the sitemap for the wiki anymore.

With webscraper, I then manually transformed the resulting CSV file into JSON using the files in this repo's [editor folder](https://github.com/kevinta893/factorio-recipes-json/tree/master/editor). Sometimes it was easier to edit the CSV in excel and then transform it into JSON.

As you may already know, it might be easier to develop a tool to extract the game's data using the game's LUA files. There is a "recipes.lua" file in the game's files which is a good starting point. Though I had to add in some of my own "recipes" in this database to make recursion stop at end points like Ores. Mining ores isn't really a recipe, but it makes sense to add when traversing the recipe tree.

In terms of maintenance, I'm going to keep this repository in as a "wiki"-style database. But you're more than welcome to use what I have to make an update.

**Comment by IvanMurzak** [comment]
Yes, please. Could you update the JSON database of recipes (with craft duration as you have now) to 0.17 version. I want to developer blueprint generator for production facilities. I need the database so much.

**Comment by kevinta893** [comment]
Ok, I will begin work on getting the DB up to speed since Factorio 0.17 is now in stable release. I'll be targeting 0.17.74 (at the time of writing).

Going forward I'm going to start using versioning to distribute the database so our projects don't break at the last moment. I'll have all the previous versions published and then a "latest" for those who like living on the edge. Any inconsistencies (like errors from that particular version of Factorio) will still be updated. I'll look into doing some CDN.

I'll also try to post the data scraping tools I used to extract from wiki.factorio.com. That way contributors can independently validate the data or make their own spin. I might look into the game files again too. Although I would like to double check with getting permissions from the wiki or the game developers.

Slowly but surely, I will get to this project. I appreciate your patience.
